### DISCUSSION TITLE
Security of blockchain scenarios, what do you think?

### DISCUSSION THREAD

**Discussion by jzakotnik** [openkfw/TruBudget#1442] [Q&A]
I think we don't have detailed technical documentation on how the consensus in the Trubudget chain exactly technically works and what would be needed to hack and rewrite the blockchain. At the moment everyone has to **trust** that the technology somehow works and that it is bitcoin core compatible.

Assuming the following scenario:
1 donor has 1 TB node and the partner has 2 TB nodes. What would be the process to change the blockchain on the partner side e.g. 1 week after data was entered? Would this be noted? What is the effort to change it after 30min or after 6 months? How do the various parameters (e.g. nr of confirmations) play into this? I assume one could fork another chain at some point, copy the data over and at some point the original chain would be discarded?

Anyone an idea how this could be made transparent or visualised? Another option would be a hackathon to actually change data in a sample installation?

**Comment by fretchen** [comment]
Thanks. I am super curious to find out where such tests take us. As for the visualization I really like the [mermaid.js](http://mermaid.js.org/config/Tutorials.html) library for such kind of diagrams.
